src/components/VariantSearch/VariantSearch.tsx

```diff
@@ -124,50 +130,146 @@ export const VariantSearch = ({ onVariantSelect }: Props) => {
     return suggestions.slice(0, 20);
   };
 
+  const handleInputChange = (newValue: string, change: InputActionMeta) => {
+    if(change.action === 'input-change') {
+      // when input value has "," in the string
+      if (newValue.includes(",")) {
+        handleCommaSeparatedInput(newValue);
+      } else {
+        setInputValue(newValue);
+        setMenuIsOpen(true);
+      }
+    }
+
+    if(change.action === 'menu-close' || change.action === 'input-blur'){
+      setMenuIsOpen(false);
+    }
+  };
+
+  const handleCommaSeparatedInput = (inputValue: string) => {
+    const inputValues = inputValue.split(",")
+    let newSelectedOptions: SearchOption[] = []
+    let invalidQueries = ''
+    for (let query of inputValues) {
+      // remove spaces from both ends of the query
+      query = query.trim()
+      const suggestions = suggestOptions(query);
+      const selectedOption = suggestions.find(option => option.value.toUpperCase() === query.toUpperCase());
+      // If query gets back valid a valid suggest option which is not yet selected,
+      // then add the query's suggest option to the newSelectedOptions to be added to selectedOptions.
+      // Otherwise, append the query to the invalid queries string
+      if (suggestions && suggestions.length > 0
+          && selectedOption && !selectedOptions.find(option => option.value === selectedOption.value)) {
+        if(selectedOption.type === 'mutation'
+            // if there are more than one pangolin-lineage options in the comma separated input, we only accept the first one
+            || (selectedOption.type === 'pangolin-lineage' && newSelectedOptions.filter(option => option.type === 'pangolin-lineage').length < 1)) {
+          newSelectedOptions.push(selectedOption);
+        }
+      }else{
+        invalidQueries += query + ','
+      }
+    }
+
+     if (newSelectedOptions.length > 0) {
+       setSelectedOptions([...selectedOptions, ...newSelectedOptions]);
+     }
+
+    // remove the last "," in the invalidQueries string
+    invalidQueries = invalidQueries.slice(0, -1)
+    setInputValue(invalidQueries);
+    // leave options menu open if there are invalid queries from the list
+    // this will show the "No Options" in the menu
+    setMenuIsOpen(invalidQueries !== '');
+  }
+
+  const promiseOptions = (inputValue : string) => {
+    // resets the options to default (where input value is '') when menu is closed
+    if(!menuIsOpen){
+      inputValue = ''
```

Comment:
Why not `setInputValue('')`?

Should we use `undefined` by default when no input value? (I don't know)

Reply:
`setInputValue('')` is called right after at line 205. But because the state change is async and doesn't happen immediately so I still passed the inputValue at line 208. This code is simulating the action that when an option is selected the option menu gets closed and the suggest options get reloaded (it does not happen automatically for selecting from using comma-separated list because onChange is not fired). 

https://github.com/cevo-public/cov-spectrum-website/blob/8944d23088d2d9ce0a803f4003ea52d05d54092e/src/components/VariantSearch/VariantSearch.tsx#L201-L208

The reason why I used '' instead of undefined is because  line 262, the defaultOptions come from '' input. 

https://github.com/cevo-public/cov-spectrum-website/blob/8944d23088d2d9ce0a803f4003ea52d05d54092e/src/components/VariantSearch/VariantSearch.tsx#L262